### PR TITLE
Fix the race condition in net.go where accessing node property w/o lock

### PR DESCRIPTION
**Pull request by zuwliu**

### CODE REVIEW

File: net.go [hunk 1]
```diff
@@ -686,12 +686,17 @@ func (m *Memberlist) rawSendMsgPacket(addr string, node *Node, msg []byte) error
...
-	if node != nil && node.PMax >= 5 {
-		crc := crc32.ChecksumIEEE(msg)
-		header := make([]byte, 5, 5+len(msg))
-		header[0] = byte(hasCrcMsg)
-		binary.BigEndian.PutUint32(header[1:], crc)
-		msg = append(header, msg...)
+	if node != nil {
+		m.nodeLock.RLock()
+		pMax := node.PMax
+		m.nodeLock.RUnlock()
```
Comment (rchincha) [inline net.go:692]: Is this correct? node.PMax value may become different right after this unlock, so you may have a stale value in pMax?

Comment (zuwliu) [reply net.go:692]: That is fine, protocol is usually defined in the config file and never change during serf. Of course someone can find a way to updated it, but it is not a concern. 

I want the fix to be safe, so try to lock as less code as possible

Comment (rchincha) [reply net.go]: don't see anything that is blocking in this function - so just do a defer RUnlock()

**Review by rchincha — APPROVED** [review]
LGTM

**Pull request merged**